spec/gcs_bucket_remote_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+# rubocop:disable RSpec/FilePath, Lint/MissingCopEnableDirective
+
+require 'spec_helper'
+require 'google/cloud/storage'
+
+class RemoteGCSCloud < AssetCloud::Base
+  attr_accessor :gcs_connection
+  bucket :tmp, AssetCloud::GCSBucket
+
+  def gcs_bucket
+    gcs_connection.bucket(ENV['GCS_BUCKET'])
+  end
+end
+
+describe AssetCloud::GCSBucket, if: ENV['GCS_PROJECT_ID'] && ENV['GCS_KEY_FILEPATH'] && ENV['GCS_BUCKET'] do
+  directory = File.dirname(__FILE__) + '/files'
+
+  before(:all) do
+    @cloud = RemoteGCSCloud.new(directory, 'assets/files')
+
+    @cloud.gcs_connection = Google::Cloud::Storage.new(
+        project_id: ENV['GCS_PROJECT_ID'],
+        credentials: ENV['GCS_KEY_FILEPATH']
+      )
+    @bucket = @cloud.buckets[:tmp]
+  end
+
+  after(:all) do
+    @bucket.clear
+  end
+
+  it "#ls calls files with no arguments if no key is provided" do
```

Comment:
**nit** this test's description is very specific to the internals of the class, would it make sense to rename to something like 

`it "#ls with no arguments returns all files in the bucket" do`

Also, there's a test missing for `ls with arguments`